library/Vanilla/Authenticator/SSOAuthenticator.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace Vanilla\Authenticator;
+
+use Garden\Web\RequestInterface;
+use Vanilla\Models\SSOInfo;
+
+abstract class SSOAuthenticator extends Authenticator {
+    /**
+     * Tells whether the data returned by this authenticator is authoritative or not.
+     * User info/roles can only be synchronized by trusted authenticators.
+     *
+     * @var bool
+     */
+    private $isTrusted = false;
+
+    /**
+     * Authenticator constructor.
+     *
+     * @param string $authenticatorID Currently maps to "UserAuthenticationProvider.AuthenticationKey".
+     */
+    public function __construct($authenticatorID) {
```

Comment:
You should be able to ditch this, since all your doing is calling the parent constructor.

Reply:
The docblock is important so it has to stay.